File: src/models/entradaProd.java
package models;

import java.sql.Date;

public class entradaProd {

	public entradaProd(String asunto, String fecha,  String area, boolean confidencial, boolean urgente) {
		super();
		this.asunto = asunto;
		this.fecha = fecha;
		this.area = area;
		this.confidencial = confidencial;
		this.urgente = urgente;
		this.jefe1 = 0;
		this.jefe2 = 0;
		this.jefe3 = 0;
		this.jefe4 = 0;
		this.jefe5 = 0;
		this.observaciones = "";
		this.soloCoronel = false;
		this.tramitado = false;
	}
	
	int id;
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getAsunto() {
		return asunto;
	}
	public void setAsunto(String asunto) {
		this.asunto = asunto;
	}
	public String getFecha() {
		return fecha;
	}
	public void setFecha(String fecha) {
		this.fecha = fecha;
	}
	public String getFile() {
		return file;
	}
	
	public String getArea() {
		return area;
	}
	
	public void setArea(String area) {
		this.area = area;
	}
	public void setFile(String file) {
		this.file = file;
	}

	
	public int getJefe1() {
		return jefe1;
	}
	public void setJefe1(int jefe1) {
		this.jefe1 = jefe1;
	}
	public int getJefe2() {
		return jefe2;
	}
	public void setJefe2(int jefe2) {
		this.jefe2 = jefe2;
	}
	public int getJefe3() {
		return jefe3;
	}
	public void setJefe3(int jefe3) {
		this.jefe3 = jefe3;
	}
	public int getJefe4() {
		return jefe4;
	}
	public void setJefe4(int jefe4) {
		this.jefe4 = jefe4;
	}
	public int getJefe5() {
		return jefe5;
	}
	public void setJefe5(int jefe5) {
		this.jefe5 = jefe5;
	}
	public boolean isConfidencial() {
		return confidencial;
	}
	public void setConfidencial(boolean confidencial) {
		this.confidencial = confidencial;
	}
	public boolean isUrgente() {
		return urgente;
	}
	public void setUrgente(boolean urgente) {
		this.urgente = urgente;
	}
	public String getObservaciones() {
		return observaciones;
	}
	public void setObservaciones(String observaciones) {
		this.observaciones = observaciones;
	}
	
	public boolean isSoloCoronel() {
		return soloCoronel;
	}
	public void setSoloCoronel(boolean soloCoronel) {
		this.soloCoronel = soloCoronel;
	}
	public boolean isTramitado() {
		return tramitado;
	}
	public void setTramitado(boolean tramitado) {
		this.tramitado = tramitado;
	}
	
	String asunto;
	String fecha;
	String file;
	String area;
	String observaciones;
	
	

	boolean urgente;
	boolean confidencial;
	boolean soloCoronel;
	boolean tramitado;

	

	int jefe1;
	int jefe2;
	int jefe3;
	int jefe4;
	int jefe5;

	
	public boolean isJefe1() {
		// TODO Auto-generated method stub
		if (jefe1 == 1 || jefe1 == 2) {
			return true;
		}else return false;
	}
	public boolean isJefe2() {
		// TODO Auto-generated method stub
		if (jefe2 == 1 || jefe2 == 2) {
			return true;
		}else return false;
	}
	public boolean isJefe3() {
		// TODO Auto-generated method stub
		if (jefe3 == 1 || jefe3 == 2) {
			return true;
		}else return false;
	}
	public boolean isJefe4() {
		// TODO Auto-generated method stub
		if (jefe4 == 1 || jefe4 == 2) {
			return true;
		}else return false;
	}
	public boolean isJefe5() {
		// TODO Auto-generated method stub
		if (jefe5 == 1 || jefe5 == 2) {
			return true;
		}else return false;
	}
	
}

File: src/models/entradaProd.js
class EntradaProd {
    constructor(asunto, fecha, area, confidencial, urgente) {
        this.id = null; // Or generate a unique ID
        this.asunto = asunto;
        this.fecha = fecha;
        this.file = null; // Initialize as null or appropriate default
        this.area = area;
        this.observaciones = "";
        this.urgente = urgente;
        this.confidencial = confidencial;
        this.soloCoronel = false;
        this.tramitado = false;
        this.jefe1 = 0;
        this.jefe2 = 0;
        this.jefe3 = 0;
        this.jefe4 = 0;
        this.jefe5 = 0;
    }

    // Getters
    getId() { return this.id; }
    getAsunto() { return this.asunto; }
    getFecha() { return this.fecha; }
    getFile() { return this.file; }
    getArea() { return this.area; }
    getObservaciones() { return this.observaciones; }
    isUrgente() { return this.urgente; }
    isConfidencial() { return this.confidencial; }
    isSoloCoronel() { return this.soloCoronel; }
    isTramitado() { return this.tramitado; }
    getJefe1() { return this.jefe1; }
    getJefe2() { return this.jefe2; }
    getJefe3() { return this.jefe3; }
    getJefe4() { return this.jefe4; }
    getJefe5() { return this.jefe5; }

    // Setters
    setId(id) { this.id = id; }
    setAsunto(asunto) { this.asunto = asunto; }
    setFecha(fecha) { this.fecha = fecha; }
    setFile(file) { this.file = file; }
    setArea(area) { this.area = area; }
    setObservaciones(observaciones) { this.observaciones = observaciones; }
    setUrgente(urgente) { this.urgente = urgente; }
    setConfidencial(confidencial) { this.confidencial = confidencial; }
    setSoloCoronel(soloCoronel) { this.soloCoronel = soloCoronel; }
    setTramitado(tramitado) { this.tramitado = tramitado; }
    setJefe1(jefe1) { this.jefe1 = jefe1; }
    setJefe2(jefe2) { this.jefe2 = jefe2; }
    setJefe3(jefe3) { this.jefe3 = jefe3; }
    setJefe4(jefe4) { this.jefe4 = jefe4; }
    setJefe5(jefe5) { this.jefe5 = jefe5; }

    // Methods like isJefeX()
    isJefe1() { return this.jefe1 === 1 || this.jefe1 === 2; }
    isJefe2() { return this.jefe2 === 1 || this.jefe2 === 2; }
    isJefe3() { return this.jefe3 === 1 || this.jefe3 === 2; }
    isJefe4() { return this.jefe4 === 1 || this.jefe4 === 2; }
    isJefe5() { return this.jefe5 === 1 || this.jefe5 === 2; }
}

module.exports = EntradaProd;
